src/zerocash/tests/zerocashTest.cpp

```diff
@@ -210,33 +210,30 @@ void test_pour(libzerocash::ZerocashParams& p,
         libzerocash::Coin coin(addr.getPublicAddress(), *it);
 
         // commitment from coin
-        std::vector<bool> commitment(ZC_CM_SIZE * 8);
-        libzerocash::convertBytesVectorToVector(coin.getCoinCommitment().getCommitmentValue(), commitment);
+        uint256 commitment(coin.getCoinCommitment().getCommitmentValue());
 
         // insert commitment into the merkle tree
-        std::vector<bool> index;
-        merkleTree.insertElement(commitment, index);
+        merkleTree.append(commitment);
+
+        // and append to any witnesses
+        for(vector<pour_input_state>::iterator wit = input_state.begin(); wit != input_state.end(); ++wit) {
+            std::get<2>(*wit).append(commitment);
```

Comment:
It's unclear what the elements of this tuple are. Is this just something specific to this test? If it's part of the witness API, perhaps it should be replaced with a struct or class.